NewForce-Cohort-7/tabloidmvc-yellow-belly-yackers
# Ticket # 29 - Register New Users

# Description
Completed ticket # 29 in Trello - [Register](https://trello.com/c/0W0GsdLN/29-29-register)
Added a “REGISTER” link to the header Menu and a button on the Home Page (when logged out).
When the link is clicked, you are presented with a form to enter  User Profile information.

## Type of change


# Testing Instructions

After cloning this repository to your own machine:

Download these two files [01_TabloidMVC_Create_DB.sql](https://github.com/NewForce-Cohort-7/tabloidmvc-yellow-belly-yackers/blob/main/SQL/01_TabloidMVC_Create_DB.sql) and [02_TabloidMVC_Seed_Data.sql](https://github.com/NewForce-Cohort-7/tabloidmvc-yellow-belly-yackers/blob/main/SQL/02_TabloidMVC_Seed_Data.sql) to create your database. Use the two links and click the Download raw file button on the right side before the code block in github starts.

In your gitbash terminal:

Navigate to the root directory for the repo (../tabloidmvc-yellow-belly-yackers)
git fetch -a
git checkout nm-registernewusers
start TabloidMVC.sln
Once Visual Studio opens:

Click File > Open > File, and open the two .sql files you just downloaded.
Starting with the 01_TabloidMVC_Create_DB.sql, click the green, triangular, Execute button in the top left of the screen, above where the data is entered.
Repeat the process with the 02_TabloidMVC_Seed_Data.sql file.
Click the green, triangular, button labeled TabloidMVC at the top center of your Visual Studio.
The app should launch by opening a new instance of your default web browser.

## Testing the new feature:

In the header, next to the Login button, there should be a link to Register, click this.

You should be navigated to a form to create a new user.
Fill out each input field. All four of them are required.
After filling out the input fields click the Register button. 
You will then be redirected to the Home page.
To check that the user has been registered, open the SQL Server Object Explorer from the View menu in Visual Studio.
Expand the option for your SQL Server Instance.
Expand Databases.
Right click on the TabloidMVC database and select New Query.
Copy this SQL query into the field and run the query:

`SELECT u.Id, u.DisplayName, u.FirstName, u.LastName, u.Email, u.CreateDateTime, u.ImageLocation, u.UserTypeId, ut.[Name] as UserTypeName
                        FROM UserProfile u
                        LEFT JOIN UserType ut on ut.Id = u.UserTypeId`

The results should show a table with two entries, one with an Id of 1 which was part of the seed data, and the second should be the data you registered on the app. This data should have a UserTypeId of 2, and the CreateTimeDate should be the time you successfully registered the user.

# Checklist:
